src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1581,7 +1585,8 @@ StoreQueryResult(CitusScanState *scanState, MultiConnection *connection,
 			category = ERRCODE_TO_CATEGORY(ERRCODE_INTEGRITY_CONSTRAINT_VIOLATION);
 			isConstraintViolation = SqlStateMatchesCategory(sqlStateString, category);
 
-			if (isConstraintViolation || failOnError)
+			if (isConstraintViolation || failOnError ||
```

Comment:
`failOnError` is misnamed here, since there are cases that this can be `false` but we still fail. We should at least update the comments for `StoreQueryResult()` and `ConsumeQueryResult()`.